src/helpers/redirects.ts

```diff
@@ -14,9 +18,11 @@ import {
   redirectsForNextRoute,
   redirectsForNextRouteWithData,
   routeToDataRoute,
-  targetForFallback,
 } from './utils'
 
+const matchesMiddleware = (middleware: Array<string>, route: string): boolean =>
```

Comment:
```suggestion
const matchesMiddleware = (middleware: Array<string> | undefined, route: string): boolean =>
```

[sand] Or optional? I see you check the existence in the line below

Reply:
Looks like `middleware` is always defined, because `getMiddleware` defaults to `[]`, so I'll just remove the optional chain check